domain/secretbackend/state/state.go

```diff
@@ -46,66 +47,253 @@ func NewState(factory coredatabase.TxnRunnerFactory, logger Logger) *State {
 // GetModel is responsible for returning the model for the provided uuid. If no
 // model is found the given uuid then an error of
 // [github.com/juju/juju/domain/model/errors.NotFound] is returned.
-func (s *State) GetModel(ctx context.Context, uuid coremodel.UUID) (coremodel.Model, error) {
+func (s *State) GetModel(ctx context.Context, uuid coremodel.UUID) (secretbackend.Model, error) {
 	db, err := s.DB()
 	if err != nil {
-		return coremodel.Model{}, errors.Trace(err)
+		return secretbackend.Model{}, errors.Trace(err)
 	}
 
-	var m coremodel.Model
-	return m, db.StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
-		var err error
-		m, err = modelstate.Get(ctx, tx, uuid)
-		return err
+	stmt, err := s.Prepare(`
+SELECT &Model.*
+FROM v_model_secret_backend
+WHERE uuid = $M.uuid`, sqlair.M{}, Model{})
+	if err != nil {
+		return secretbackend.Model{}, errors.Trace(err)
+	}
+	var m Model
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, stmt, sqlair.M{"uuid": uuid}).Get(&m)
```

Comment:
We use sqlair maps a lot which is fine. We've also started to initialise the Model m with the uuid we want and use that as both an in and out parameter. So we only need to pass the one type `Model{}` to the prepare etc.